Sources/FaunaDB/JSON.swift

```diff
@@ -197,6 +197,32 @@ private extension JsonType {
         }
     }
 
+    private func toRefV(object: [String: JsonType]) throws -> Value {
+        guard case let .string(id)? = object["id"] else { throw JsonError.invalidRefValue }
+
+        if object["class"] == nil && object["database"] == nil {
+            return Native.fromName(id)
+        }
+
+        var clazz: RefV? = nil
+        if let cls = try object["class"]?.toValue() {
```

Comment:
Class and Database ref conversion are exactly the same. There is a little trick that you can use with `guard` if you extract this to a different function:

```swift
//...
let clazz = try object["class"].map(forceRefV)
let database = try object["database"].map(forceRefV)
//...

private func forceRefV(_ json: JsonType) throws -> RefV {
    guard let ref = try json.toValue() as? RefV else {
        throw JsonError.invalidRefValue
    }
    return ref
}
```